GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSUpgradeReminderTests.cs

```diff
@@ -62,6 +63,51 @@ public void NoReminderForLeftOverDownloads()
             this.VerifyUpgradeVerbStopsReminder();
         }
 
+        [TestCase]
+        public void UpgradeTimerScheduledOnServiceStart()
+        {
+            this.RestartService();
+            Thread.Sleep(TimeSpan.FromSeconds(15));
+
+            this.ServiceLogContainsUpgradeMessaging().ShouldBeTrue();
+        }
+
+        private bool ServiceLogContainsUpgradeMessaging()
+        {
+            string upgradeTimerMessage = "Checking for product upgrades. (Start)";
+            string serviceLogFolder = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData),
+                "GVFS",
+                GVFSServiceProcess.TestServiceName,
+                "Logs");
+            DirectoryInfo logsDirectory = new DirectoryInfo(serviceLogFolder);
+            IEnumerable<FileInfo> files = logsDirectory.GetFiles()
+                .OrderByDescending(f => f.LastWriteTime)
+                .Take(2);
+
+            foreach (FileInfo nextFile in files)
+            {
+                using (StreamReader fileStream = new StreamReader(File.Open(nextFile.FullName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite)))
+                {
+                    for (int lineCount = 0; lineCount < 20; lineCount++)

```

Comment:
Why limit the search to the first 20 lines?

Reply:
It usually appears in the first 20. But on second thought, I think this approach could make the test unnecessarily fragile. Also since we are restarting the service in the beginning of the test case, the log file itself wouldn't be very large. Reading the whole file now (only in the worst case when the upgrade check message string is missing.)